src/Handlers/SystemWeb/RequestEndExtensions.cs

```diff
@@ -0,0 +1,68 @@
+// MIT License.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.SystemWebAdapters;
+using Microsoft.AspNetCore.SystemWebAdapters.Features;
+
+namespace System.Web;
+
+#pragma warning disable SYSWEB1001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+internal static class RequestEndExtensions
+{
+    /// <summary>
+    /// By design, the System.Web adapters try not to throw when <see cref="HttpResponse.End"/> but rather tracks it in other ways.
+    /// However, WebForms was built with that expectation and some common scenarios (such as postback calls) break without this.
+    /// </summary>
+    [Diagnostics.CodeAnalysis.SuppressMessage("Reliability", "CA2007:Consider calling ConfigureAwait on the awaited task", Justification = "Need the type itself")]
+    public static IApplicationBuilder EnsureRequestEndThrows(this IApplicationBuilder app)
+        => app.Use(async (ctx, next) =>
+        {
+            await using var feature = new RequestEndThrowingFeature(ctx);
+
+            ctx.Features.Set<IHttpResponseEndFeature>(feature);
+
+            try
+            {
+                await next(ctx).ConfigureAwait(false);
+            }
+            catch (RequestEndException)
```

Comment:
do we know if Framework is doing anything with the exception or just eating it up to end the response and close any further interactions ? If later, does that mean this used to happen with every response.redirect ?

Reply:
On framework, it tends to throw ThreadAbortException which the native module will handle (can't catch it in general). Allowing the adapters to handle it with their infrastructure (they check for the IsEnded status and closes things out gracefully without needing to throw an exception).